test-harness/src/DequeueHandler.ts

```diff
@@ -0,0 +1,68 @@
+import { LambdaInterface } from "@aws-lambda-powertools/commons";
+import { Logger } from "@aws-lambda-powertools/logger";
+import { PutObjectCommand, S3Client } from "@aws-sdk/client-s3";
+import { NodeHttpHandler } from "@aws-sdk/node-http-handler";
+import { Constants } from "./utils/Constants";
+import { BatchItemFailure } from "./utils/BatchItemFailure";
+
+const POWERTOOLS_LOG_LEVEL = process.env.POWERTOOLS_LOG_LEVEL
+	? process.env.POWERTOOLS_LOG_LEVEL
+	: Constants.DEBUG;
+const POWERTOOLS_SERVICE_NAME = process.env.POWERTOOLS_SERVICE_NAME
+	? process.env.POWERTOOLS_SERVICE_NAME
+	: Constants.DEQUEUE_LOGGER_SVC_NAME;
+
+export const logger = new Logger({
+	logLevel: POWERTOOLS_LOG_LEVEL,
+	serviceName: POWERTOOLS_SERVICE_NAME,
+});
+
+export const s3Client = new S3Client({
+	region: process.env.REGION,
+	maxAttempts: 2,
+	requestHandler: new NodeHttpHandler({
+		connectionTimeout: 29000,
+		socketTimeout: 29000,
+	}),
+});
+
+class DequeueHandler implements LambdaInterface {
+	async handler(event: any): Promise<any> {
+		logger.info("Starting to process records");
+		const batchFailures: BatchItemFailure[] = [];
+
+		for await (const record of event.Records) {
+			const body = JSON.parse(record.body);
+			const bucket = process.env.EVENT_TEST_BUCKET_NAME;
+			const propertyName = process.env.PROPERTY_NAME || "sub";
+			// Accounts for values that are nested (eg user.session_id)
+			const propertyValue = propertyName
```

Comment:
Could you elaborate on why? `propertyName` is always going to be a string, so split will always work and it will always return an array for reduce to work on

Reply:
I was thinking for the entire for loop to be in try/catch but ignore if feel not needed